src/main/java/games/strategy/engine/framework/map/download/DownloadFileParser.java

```diff
@@ -26,38 +27,31 @@ private DownloadFileParser() {}
   }
 
   public static List<DownloadFileDescription> parse(final InputStream is) {
-    @SuppressWarnings("unchecked")
-    final List<Map<String, String>> yamlData = new Yaml().loadAs(is, List.class);
+    final JSONArray yamlData = new JSONArray(new Yaml().loadAs(is, List.class));
 
     final List<DownloadFileDescription> rVal = new ArrayList<>();
-    for (final Map<String, String> yaml : yamlData) {
-      final String url = yaml.get(Tags.url.toString());
-      final String description = yaml.get(Tags.description.toString());
-      final String mapName = yaml.get(Tags.mapName.toString());
-
-      Version version = null;
-      final Object versionObject = yaml.get(Tags.version.toString());
-      if (versionObject != null) {
-        version = new Version(String.valueOf(versionObject));
-      }
-
+    for (int i = 0; i < yamlData.length(); i++) {
```

Comment:
Any reason not to use a for-each loop?: `for(JSONObject yaml : yamlData)` ?

Reply:
Unfortunately, `JSONArray` implements `Iterable<Object>` not `Iterable<JSONObject>` because the elements of a JSON array can be any of multiple data types, `JSONObject` being one of them.  Thus, the for-each loop would be `for (Object yaml : yamlData)` and a cast would be required in the body of the loop.  Not sure which is better.

Reply:
What if we create a helper to do this? I think ideally we should avoid accessing `yamlData` data directly and instead put an API in front of it that would make the client code trivial. Taking that a step further we could likely convert the loop to a streaming expression.